### PR TITLE
Update the minimum external LLVM to 10

### PR DESCRIPTION
**Pull request by cuviper**
r? @nikic

### CODE REVIEW

**Comment by nikic** [comment]
Can you please also drop https://github.com/rust-lang/rust/blob/2287a8823d2d731b9bf3064da305fc5c408b24e2/compiler/rustc_llvm/build.rs#L126 and related? (This could've been dropped for LLVM 9 already.)

File: library/std/src/sys/sgx/abi/mem.rs [hunk 1]
```diff
@@ -38,8 +38,7 @@ pub fn image_base() -> u64 {
         asm!(
             "lea IMAGE_BASE(%rip), {}",
             lateout(reg) base,
-            // NOTE(#76738): ATT syntax is used to support LLVM 8 and 9.
-            options(att_syntax, nostack, preserves_flags, nomem, pure),
+            options(nostack, preserves_flags, nomem, pure),
```
Comment (nagisa) [inline library/std/src/sys/sgx/abi/mem.rs:41]: This is no longer `att_syntax`, you want something like `lea {}, [rip + IMAGE_BASE]` I think.

**Comment by cuviper** [comment]
I'll make the requested changes and run the relevant docker CI locally to make sure of the `att_syntax` removal.

@bors r-

File: library/core/src/num/dec2flt/algorithm.rs [hunk 2]
```diff
@@ -64,8 +64,7 @@ mod fpu_precision {
             asm!(
                 "fldcw ({})",
                 in(reg) &cw,
-                // FIXME: We are using ATT syntax to support LLVM 8 and LLVM 9.
-                options(att_syntax, nostack),
+                options(nostack),
```
Comment (nagisa) [inline library/core/src/num/dec2flt/algorithm.rs:67]: Hm, not super sure, but I think these want to become `fldcw [{}]` and `fnstcw [{}]` now.

**Comment by cuviper** [comment]
OK, it should be ready again.

**Comment by Amanieu** [comment]
A lot of the `asm!` tests were marked with `no-system-llvm` because they required LLVM 10. This can now be removed.

**Comment by JohnTitor** [comment]
```
curl: (56) OpenSSL SSL_read: Connection was reset, errno 10054

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
```

@bors retry

**Comment by mati865** [comment]
@Amanieu do you want to open the PR for your comment?

**Comment by Amanieu** [comment]
#83485

**Pull request merged**